draft-holmberg-mmusic-ice-premature.xml

```diff
@@ -100,15 +98,93 @@
+            its own. If the agent somehow knows that the remote endpoint is
+            directly reachable, gathering local candidates is unnecessary and
+            will only cause delays; the remote endpoint can discover the
+            appropriate local candidate via connectivity checks.
+          </t>
+          <t>
+            However, following the procedures from
+            <xref target="RFC8445"></xref> strictly will result in immediate
+            ICE failure, since the checklist at the remote endpoint will be
+            empty.
+          </t>
+        </section>
+        <section title="All Candidates Discarded">
+          <t>
+            Even if the ICE Agent provides candidates, they may be discarded
+            by the remote endpoint if it does not know what to do with them.
+            For example, candidates may use an address family that the remote
+            endpoint does not support, (e.g., a host candidate with an IPv6
+            address in a NAT64 scenario), or may not be usable for some other
+            reason (e.g., a candidate that contains a FQDN that fails to
+            resolve).
+          </t>
+          <t>
+            In these scenarios, when the candidates are discarded, the
+            checklist at the remote endpoint will once again be empty, leading
+            to immediate ICE failure.
+          </>
+        </section>
+        <section title="Immediate Candidate Pair Failure">
+          <t>
+            Section 7.2.5.2 of <xref target="RFC8445"></xref> describes several
+            situations in which a candidate pair will be considered to have
+            failed, well before the connectivity check timeout.
+          </t>
+          <t>
+            As a result, even if the ICE Agent provides usable candidates,
+            the pairs created by the remote endpoint may fail immediately when
+            checked, e.g., a check to a nonroutable address that receives an
+            immediate ICMP error.
+          </t>
+          <t>
+            In this situation, the checklist at the remote endpoint may contain
+            only failed pairs, resulting in immediate ICE failure.
+          </t>
+        </section>
+      </t>
+    </section>
+
     <section title="Update to RFC 8445">
+      <t>
+        In order to avoid this problem, the ICE Agent needs to wait enough
+        time to allow peer-reflexive candidates to be discovered. Accordingly,
+        when beginning ICE checks [add reference], the ICE Agent SHOULD set a
+        timer with duration equal to that of the ICE connectivity check timeout.
+        If, while this timer is running, a checklist has no pairs left to check,
+        i.e., there are no pairs that are not in the failed state, the ICE
+        Agent MUST not conclude that ICE processing has failed, and MUST wait
+        for the timer to elapse before doing so.
+      </t>
     </section>
```

Comment:
I need to read this text more in detail. But I am ok to accept it for now, so that we can submit the first version of the draft.